content/blog/2021/09/towards-a-universal-policy-platform.md

```diff
@@ -0,0 +1,155 @@
+
+By leveraging the power of WebAssembly, Kubewarden allows policy authors
+to write policies using traditional programming languages such as Rust, Go,
+AssemblyScript and Swift.
+
+Kubewarden policies, once compiled into WebAssembly modules, are then
+distributed using regular OCI registries. This allows Operators to have a
+consistent way to securely distribute both container images and policies.
+
+
+Kubewarden is one of the Open Source projects working to provide a
+Policy As Code solution to Kubernetes. Historically, the fist project in
+this space has been [Open Policy Agent](https://openpolicyagent.org/), also known
+as *"OPA"*.
+
+Open Policy Agent policies are written using a purpose-built query language
+called [Rego](https://www.openpolicyagent.org/docs/latest/policy-language/).
+
+We don't think there's a right or wrong approach when it comes to writing
+policies. Both programming languages (Rust, Go, ...) and query languages (Rego)
+have advantages and disadvantages.
+We strongly believe policy authors should have the freedom to pick the tool that
+best suits their need.
+
+At the same time, we realize that this flexibility can significantly complicate
+the lives of Operation teams. Simplicity is paramount when operating Kubernetes,
+having different ways to distribute and enforce policies goes into the opposite
+direction.
+
+The Kubewarden team wants to further expand the freedom of policy author, without
+having to compromise on the operation experience we currently provide.
+
+We want to provide a cohesive way to distribute policies, a uniform
+way to enforce them and a single platform to host them. All of that regardless of
+the language used to write these individual policies.
+
+We want to provide a **Universal Policy Platform**.
+
+## WebAssembly to the rescue
+
+Rego is a query language created by the Open Policy Agent project. The language
+is inspired by [Datalog](https://en.wikipedia.org/wiki/Datalog), its
+main purpose is to perform queries on some JSON input data and provide some
+output as response.
+
+The good news is that Rego programs can be built into WebAssembly modules using
+the official `opa` command line utility. This is documented in depth inside
+of the [official documentation](https://www.openpolicyagent.org/docs/latest/wasm/)
+of Open Policy Agent.
+
+While building a Rego program into WebAssembly is simple, running the resulting WebAssembly
+module requires more efforts. The good news is... we did the hard work!
+
+We wrote a Rust library that can be used to invoke Rego policies compiled into
+WebAssembly modules.
+This library is called Burrego (a burrito with a Rego filling... obvious isn't it?! 🌯 🤓)
+and can be found [here](https://github.com/kubewarden/policy-evaluator/tree/main/crates/burrego).
+
+The Rego language provides some [built-in functions](https://www.openpolicyagent.org/docs/latest/policy-reference/#built-in-functions)
+to perform String, Encoding, Regex, ... operations.
```

Comment:
I find `... operations` strange here. I think the last part of the sentence could be rephrased.

Reply:
I changed it